topic0/src/abstract_factory/DatabaseFactory.java

```diff
@@ -0,0 +1,7 @@
+package abstract_factory;
+
+public abstract class DatabaseFactory {
+	abstract MysqlDatabase getMysql(String mysql);
```

Comment:
this is not who the pattern works, because the final user of these factories should'n be using methods tied to final implementations, at the end that's what you want, you want to have your code decoupled from final implementation, 
DatabaseFactory databaseConnection = ConnectionMaker.establishConnection(someUnknownString);
Database database = databaseConnection.getConnection(cnxType)

this way you can code without taking care of what kind of database you've used and not only the type (Mongo or MySQl) but also if it is a client or accounts